[Bug]: puppeteer 19.9.* cannot be installed behind a corporate proxy

### Bug expectation

I expected to install puppeteer 19.* for Node.js 14.20.1 on Red Hat Enterprise Linux Server release 7.9 (Maipo) behind a corporate proxy.

What happened instead was the error below.
- The Chromium binary can be downloaded from https://storage.googleapis.com/chromium-browser-snapshots/Linux_x64/1108766/chrome-linux.zip using wget or curl
- the proxy environmental variables have been correctly set. 
- A change has been made from puppeteer 19.9.* and up so that this no longer works.
- puppeteer 19.8* can be installed as expected.

### Bug behavior

- [ ] Flaky
- [ ] PDF

### Minimal, reproducible example

```TypeScript
> npm install puppeteer@19.9.*

npm ERR! code 1
npm ERR! path [edit]/node_modules/puppeteer
npm ERR! command failed
npm ERR! command sh -c node install.js
npm ERR! ERROR: Failed to set up Chromium r1108766! Set "PUPPETEER_SKIP_DOWNLOAD" env variable to skip download.
npm ERR! Error: Download failed: server returned code 407. URL: https://storage.googleapis.com/chromium-browser-snapshots/Linux_x64/1108766/chrome-linux.zip
npm ERR!     at [edit]/node_modules/@puppeteer/browsers/lib/cjs/httpUtil.js:121:31
npm ERR!     at ClientRequest.requestCallback ([edit]/node_modules/@puppeteer/browsers/lib/cjs/httpUtil.js:98:13)
npm ERR!     at Object.onceWrapper (events.js:520:26)
npm ERR!     at ClientRequest.emit (events.js:400:28)
npm ERR!     at HTTPParser.parserOnIncomingClient (_http_client.js:647:27)
npm ERR!     at HTTPParser.parserOnHeadersComplete (_http_common.js:127:17)
npm ERR!     at Socket.socketOnData (_http_client.js:515:22)
npm ERR!     at Socket.emit (events.js:400:28)
npm ERR!     at addChunk (internal/streams/readable.js:293:12)
npm ERR!     at readableAddChunk (internal/streams/readable.js:267:9)

 
npm ERR! A complete log of this run can be found in: [edit]/.npm/_logs/2023-05-03T20_23_05_747Z-debug-0.log
```


### Error string

ERR! Error: Download failed: server returned code 407.

### Puppeteer configuration

_No response_

### Puppeteer version

19.9.*

### Node version

14.10.1

### Package manager

npm

### Package manager version

9.6.5

### Operating system

Linux

--- a/packages/browsers/package.json
+++ b/packages/browsers/package.json
@@ -99,10 +99,8 @@
   "dependencies": {
     "debug": "4.3.4",
     "extract-zip": "2.0.1",
-    "http-proxy-agent": "6.1.0",
-    "https-proxy-agent": "5.0.1",
     "progress": "2.0.3",
-    "proxy-from-env": "1.1.0",
+    "proxy-agent": "6.2.0",
     "tar-fs": "2.1.1",
     "unbzip2-stream": "1.4.3",
     "yargs": "17.7.1"

--- a/packages/browsers/src/httpUtil.ts
+++ b/packages/browsers/src/httpUtil.ts
@@ -19,12 +19,7 @@ import * as http from 'http';
 import * as https from 'https';
 import {URL} from 'url';
 
-import * as httpProxyAgent from 'http-proxy-agent';
-import * as httpsProxyAgent from 'https-proxy-agent';
-import {getProxyForUrl} from 'proxy-from-env';
-
-const {HttpProxyAgent} = httpProxyAgent;
-const {HttpsProxyAgent} = httpsProxyAgent;
+import {ProxyAgent} from 'proxy-agent';
 
 export function headHttpRequest(url: URL): Promise<boolean> {
   return new Promise(resolve => {
@@ -55,23 +50,8 @@ export function httpRequest(
     path: url.pathname + url.search,
     method,
     headers: keepAlive ? {Connection: 'keep-alive'} : undefined,
+    agent: new ProxyAgent(),
   };
-
-  const proxyURL = getProxyForUrl(url.toString());
-  if (proxyURL) {
-    const proxy = new URL(proxyURL);
-    if (proxy.protocol === 'http:') {
-      options.agent = new HttpProxyAgent(proxyURL);
-    } else {
-      options.agent = new HttpsProxyAgent({
-        host: proxy.host,
-        path: proxy.pathname,
-        port: proxy.port,
-        secureProxy: proxy.protocol === 'https:',
-        headers: options.headers,
-      });
-    }
-  }
 
   const requestCallback = (res: http.IncomingMessage): void => {
     if (

